packages/core/src/browser/widgets/select-component.tsx

```diff
@@ -0,0 +1,301 @@
+        if (maxWidth) {
+            maxWidth -= 10; // Decrease width by 10 due to side padding
+        }
+        const descriptionHeight = measureTextHeight(this.props.options.map(e => e.description || ''), maxWidth) + 18;
+        const singleLineHeight = measureTextHeight(firstLine.label || firstLine.value || firstLine.detail || '') + 6;
+        const optimal = descriptionHeight + singleLineHeight * this.props.options.length;
+        return optimal + 20; // Just to be safe, add another 20 pixels here
+    }
+
+    attachListeners(): void {
+        const hide = () => {
+            this.hide(true);
+        };
+        this.mountedListeners.set('scroll', hide);
+        this.mountedListeners.set('wheel', hide);
+
+        let parent = this.fieldRef.current?.parentElement;
+        while (parent) {
+            // Workaround for perfect scrollbar, since using `overflow: hidden`
+            // neither triggers the `scroll`, `wheel` nor `blur` event
+            if (parent.classList.contains('ps')) {
+                parent.addEventListener('ps-scroll-y', hide);
+            }
+            parent = parent.parentElement;
+        }
+
+        for (const [key, listener] of this.mountedListeners.entries()) {
+            window.addEventListener(key, listener);
+        }
+    }
+
+    override componentWillUnmount(): void {
+        if (this.mountedListeners.size > 0) {
+            const eventListener = this.mountedListeners.get('scroll')!;
+            let parent = this.fieldRef.current?.parentElement;
+            while (parent) {
+                parent.removeEventListener('ps-scroll-y', eventListener);
+                parent = parent.parentElement;
+            }
+            for (const [key, listener] of this.mountedListeners.entries()) {
+                window.removeEventListener(key, listener);
+            }
+        }
+    }
+
+    override render(): React.ReactNode {
+        const { options } = this.props;
+        let { selected } = this.state;
+        while (options[selected]?.separator) {
+            selected = (selected + 1) % this.props.options.length;
+        }
+        const selectedItemLabel = options[selected].label ?? options[selected].value;
+        return <>
+            <div
+                key="select-component"
+                ref={this.fieldRef}
+                tabIndex={0}
+                className="theia-select-component"
+                onClick={e => this.handleClickEvent(e)}
+                onBlur={() => this.hide(true)}
+                onKeyDown={e => this.handleKeypress(e)}
+            >
+                <div key="label" className="theia-select-component-label">{selectedItemLabel}</div>
+                <div key="icon" className={`theia-select-component-chevron ${codicon('chevron-down')}`} />
+            </div>
+            {ReactDOM.createPortal(this.renderDropdown(), this.dropdownElement)}
+        </>;
+    }
+
+    protected handleKeypress(ev: React.KeyboardEvent<HTMLDivElement>): void {
```

Comment:
With this code, <kbd>Up</kbd> and <kbd>Down</kbd> change the selection but don't reveal the select options. In VSCode, the first stroke opens the options, and subsequent <kbd>Up</kbd>s or <kbd>Down</kbd>s change selection.

Reply:
In my VSCode instance, using `Up` or `Down` with the select field focused only iterates through the preferences (i.e. previous or next preference) and does not open the dropdown field. Is it different for you?

Reply:
That's odd. Here's what I see:

![select-screencast](https://user-images.githubusercontent.com/62660806/161998742-7345b2bb-4aa2-4491-a472-b2bf1129ab8b.gif)

Reply:
Alright, I will keep it in mind for a future contribution, but for now I would keep it as it is 👍